src/main/kotlin/eu/long1/flutter/i18n/workers/I18nFileGenerator.kt

```diff
@@ -341,11 +337,27 @@ import 'package:flutter/material.dart';
 """
 
         private const val sClassHeader =
-            """class S implements WidgetsLocalizations {
+            """typedef void LocaleChangeCallback(Locale locale);
+
+class S implements WidgetsLocalizations {
   const S();
 
   static S current;
 
+  static Locale _locale;
+  static bool _shouldReload = false;
+
+  static set locale(Locale _newLocale) {
+    if (S.delegate
+            ._isSupported(_newLocale, _newLocale.countryCode?.isNotEmpty) &&
+        getLang(S._locale) != getLang(_newLocale)) {
+      _shouldReload = true;
+      S._locale = _newLocale;
+    }
+  }
+
+  static LocaleChangeCallback onLocaleChanged;
+
```

Comment:
I suggest rewriting like so:

```dart
  static set locale(Locale _newLocale) {
    _shouldReload = S.delegate._isSupported(_newLocale, _newLocale.countryCode?.isNotEmpty) &&
      getLang(S._locale) != getLang(_newLocale);
    if (_shouldReload) {
      S._locale = _newLocale;
    }
  }
```

Reply:
Absolutely. It is more clean. I'll make the commit.